lib/modules/datasource/go/base.ts

```diff
@@ -53,120 +53,138 @@ export class BaseGoDatasource {
-          ?.httpsRegExpName;
-      const gitlabModuleName =
-        BaseGoDatasource.gitlabRegExp.exec(goModule)?.groups?.regExpPath;
-      if (gitlabUrl && gitlabUrlName) {
-        if (gitlabModuleName?.startsWith(gitlabUrlName)) {
-          if (gitlabModuleName.includes('.git')) {
-            return {
-              datasource: GitlabTagsDatasource.id,
-              registryUrl: gitlabUrl,
-              packageName: gitlabModuleName.substring(
-                0,
-                gitlabModuleName.indexOf('.git')
-              ),
-            };
-          }
+    if (!sourceMatch) {
+      return null;
+    }
+    const [, prefix, goSourceUrl] = sourceMatch;
+    if (!goModule.startsWith(prefix)) {
+      logger.trace({ goModule }, 'go-source header prefix not match');
+      return null;
+    }
+    logger.debug({ goModule, goSourceUrl }, 'Go lookup source url');
+    if (goSourceUrl?.startsWith('https://github.com/')) {
+      return {
+        datasource: GithubTagsDatasource.id,
+        packageName: goSourceUrl
+          .replace('https://github.com/', '')
+          .replace(regEx(/\/$/), ''),
+        registryUrl: 'https://github.com',
+      };
+    }
+    const gitlabUrl =
+      BaseGoDatasource.gitlabHttpsRegExp.exec(goSourceUrl)?.groups
+        ?.httpsRegExpUrl;
+    const gitlabUrlName =
+      BaseGoDatasource.gitlabHttpsRegExp.exec(goSourceUrl)?.groups
+        ?.httpsRegExpName;
+    const gitlabModuleName =
+      BaseGoDatasource.gitlabRegExp.exec(goModule)?.groups?.regExpPath;
+    if (gitlabUrl && gitlabUrlName) {
+      if (gitlabModuleName?.startsWith(gitlabUrlName)) {
+        if (gitlabModuleName.includes('.git')) {
           return {
             datasource: GitlabTagsDatasource.id,
             registryUrl: gitlabUrl,
-            packageName: gitlabModuleName,
+            packageName: gitlabModuleName.substring(
+              0,
+              gitlabModuleName.indexOf('.git')
+            ),
           };
         }
-
         return {
           datasource: GitlabTagsDatasource.id,
           registryUrl: gitlabUrl,
-          packageName: gitlabUrlName,
+          packageName: gitlabModuleName,
         };
       }
 
-      const opts = hostRules.find({
-        hostType: PlatformId.Gitlab,
-        url: goSourceUrl,
-      });
-      if (opts.token) {
-        // get server base url from import url
-        const parsedUrl = URL.parse(goSourceUrl);
+      return {
+        datasource: GitlabTagsDatasource.id,
+        registryUrl: gitlabUrl,
+        packageName: gitlabUrlName,
+      };
+    }
 
-        // TODO: `parsedUrl.pathname` can be undefined
-        const packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
+    const opts = hostRules.find({
+      hostType: PlatformId.Gitlab,
+      url: goSourceUrl,
+    });
+    if (opts.token) {
+      // get server base url from import url
+      const parsedUrl = URL.parse(goSourceUrl);
```

Comment:
shouldn't we use our url utils here? or is it copy and paste?

Reply:
This is a copy/paste from the old, I confirmed by hiding white space when looking at the diff. Same as below